internal/backends/winit/event_loop.rs

```diff
@@ -600,8 +606,10 @@ pub fn run(quit_behavior: i_slint_core::backend::EventLoopQuitBehavior) {
                 }
 
                 winit::event::Event::UserEvent(CustomEvent::UpdateWindowProperties(window_id)) => {
-                    if let Some(window) = window_by_id(window_id) {
-                        window.runtime_window().update_window_properties();
+                    if let Err(insert_pos) =
+                        windows_with_pending_property_updates.binary_search(&window_id)
```

Comment:
What is the point of ordering the array by id?

Reply:
It's my habit of defensive programming :). If somehow two user events arrive with the same window id, I don't want to call update_window_properties() twice. Hence the sorting and binary search to avoid duplicates.

Reply:
But you insert and not replace the ids. so you still get duplicates in that scenario. they are just next to each other.

Reply:
`binary_search` returns `Ok(usize)` if the id exists, and it returns `Err(insert_pos)` when it's not found yet. So when an id exists, nothing is inserted. If it doesn't exist yet, it's inserted at the sorted position.